src/lib/js/array.rs

```diff
@@ -435,6 +435,46 @@ pub fn find(this: &Value, args: &[Value], interpreter: &mut Interpreter) -> Resu
     Ok(Gc::new(ValueData::Undefined))
 }
 
+/// Array.prototype.slice ( [begin[, end]] )
+///
+/// The slice method takes two arguments, start and end, and returns an array containing the
+/// elements of the array from element start up to, but not including, element end (or through the
+/// end of the array if end is undefined). If start is negative, it is treated as length + start
+/// where length is the length of the array. If end is negative, it is treated as length + end where
+/// length is the length of the array.
+/// <https://tc39.es/ecma262/#sec-array.prototype.slice>
+pub fn slice(this: &Value, args: &[Value], interpreter: &mut Interpreter) -> ResultValue {
+    let new_array = make_array(&to_value(Object::default()), &[], interpreter)?;
+    let len: i32 =
+        from_value(this.get_field_slice("length")).expect("Could not convert argument to i32");
+
+    let start = match args.get(0) {
+        Some(v) => from_value(v.clone()).expect("failed to parse argument for Array method"),
+        None => return Ok(this.clone()),
+    };
+    let end = match args.get(1) {
+        Some(v) => from_value(v.clone()).expect("failed to parse argument for Array method"),
+        None => len,
+    };
+
+    let from: i32 = if start < 0 {
+        max(len.wrapping_add(start), 0)
+    } else {
+        min(start, len)
+    };
+    let to: i32 = if end < 0 {
+        max(len.wrapping_add(end), 0)
+    } else {
+        min(end, len)
+    };
+
+    let span = max(to.wrapping_sub(from), 0);
+    for i in from..from.wrapping_add(span) {
+        add_to_array_object(&new_array, &[this.get_field(&i.to_string())])?;
```

Comment:
I think it's better to just call `this.set_field` rather than constructing a slice and calling `add_to_array_object`.

Reply:
@muskuloes Sorry, not `this.set_field` but `new_array.set_field`